oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/ebpf/analyze/EBPFProfilingAnalyzer.java

```diff
@@ -0,0 +1,157 @@
+            try {
+                return getDataDAO().queryData(taskId, r.minTime, r.maxTime);
+            } catch (IOException e) {
+                LOGGER.warn(e.getMessage(), e);
+                return Collections.<EBPFProfilingDataRecord>emptyList();
+            }
+        }).flatMap(Collection::stream).map(e -> {
+            try {
+                return EBPFProfilingStack.deserialize(e);
+            } catch (Exception ex) {
+                LOGGER.warn("could not deserialize the stack", ex);
+                return null;
+            }
+        }).filter(Objects::nonNull).distinct().collect(Collectors.toList());
+
+        // analyze
+        final List<EBPFProfilingTree> trees = analyze(stacks);
+        if (trees != null) {
+            analyzation.getTrees().addAll(trees);
+        }
+
+        return analyzation;
+    }
+
+    protected List<TimeRange> buildTimeRanges(List<EBPFProfilingAnalyzeTimeRange> timeRanges) {
+        return timeRanges.parallelStream()
+                .map(r -> buildTimeRanges(r.getStart(), r.getEnd()))
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+
+    protected List<TimeRange> buildTimeRanges(long start, long end) {
+        if (start >= end) {
+            return null;
+        }
+
+        // include latest millisecond
+        end += 1;
+
+        final List<TimeRange> timeRanges = new ArrayList<>();
+        do {
+            long batchEnd = Math.min(start + FETCH_FETCH_DURATION, end);
+            timeRanges.add(new TimeRange(start, batchEnd));
+            start = batchEnd;
+        }
+        while (start < end);
+
+        return timeRanges;
+    }
+
+    /**
+     * Analyze records
+     */
+    protected List<EBPFProfilingTree> analyze(List<EBPFProfilingStack> stacks) {
+        if (CollectionUtils.isEmpty(stacks)) {
+            return null;
+        }
+
+        // using parallel stream
+        Map<EBPFProfilingStack.Symbol, EBPFProfilingTree> stackTrees = stacks.parallelStream()
+                                                         // stack list cannot be empty
+                                                         .filter(s -> CollectionUtils.isNotEmpty(s.getSymbols()))
+                                                         .collect(Collectors.groupingBy(s -> s.getSymbols()
+                                                                                              .get(0), ANALYZE_COLLECTOR));
+
+        return new ArrayList<>(stackTrees.values());
+    }
+
+    protected IEBPFProfilingDataDAO getDataDAO() {
+        if (dataDAO == null) {
+            dataDAO = moduleManager.find(StorageModule.NAME)
+                                                         .provider()
+                                                         .getService(IEBPFProfilingDataDAO.class);
+        }
+        return dataDAO;
+    }
+
+    private static class TimeRange {
```

Comment:
What does TimeRange mean?

Reply:
The `TimeRange` using to split the query, if the analysis time range is bigger than 2 minutes, then split them.

Reply:
Then please add comments. A lot of codes should be explained in comments.

Reply:
Added.